src/models/multiBarChart.js

```diff
@@ -163,22 +188,58 @@ nv.models.multiBarChart = function() {
             // Setup Scales
             x = multibar.xScale();
             y = multibar.yScale();
+            x2 = x2Axis.scale();
+            y2 = multibar2.yScale();
+
+            x.range([0, availableWidth]);
+            x2.range([0, availableWidth]);
+
+            var xDomainMin = d3.min(data[0].values, function(d) { return d.x; });
+            var xDomainMax = d3.max(data[0].values, function(d) { return d.x; });
+
+            if(isNaN(xDomainMin)){
+                focusEnable = false;
+            }
+
+            if(focusEnable){
+                x.domain([
+                    xDomainMin,
+                    xDomainMax
+                ]);
+                x2.domain(x.domain());
+            }
 
             // Setup containers and skeleton of chart
             var wrap = container.selectAll('g.nv-wrap.nv-multiBarWithLegend').data([data]);
             var gEnter = wrap.enter().append('g').attr('class', 'nvd3 nv-wrap nv-multiBarWithLegend').append('g');
             var g = wrap.select('g');
 
-            gEnter.append('g').attr('class', 'nv-x nv-axis');
-            gEnter.append('g').attr('class', 'nv-y nv-axis');
-            gEnter.append('g').attr('class', 'nv-barsWrap');
-            gEnter.append('g').attr('class', 'nv-legendWrap');
-            gEnter.append('g').attr('class', 'nv-controlsWrap');
+            var focusEnter = gEnter.append('g').attr('class', 'nv-focus');
```

Comment:
If focusEnabled is false, these elements still get generated and appended to the DOM.